modules/xmpp/Polls.ts

```diff
@@ -0,0 +1,110 @@
+import { XMPPEvents } from '../../service/xmpp/XMPPEvents';
+
+import ChatRoom from './ChatRoom';
+import XMPP from './xmpp';
+
+export const COMMAND_ANSWER_POLL = 'answer-poll';
+export const COMMAND_NEW_POLL = 'new-poll';
+export const COMMAND_OLD_POLLS = 'old-polls';
+
+export default class Polls {
+    private _mainRoom: ChatRoom;
+    private _xmpp: XMPP;
+
+    /**
+     * Constructs polls for a room.
+     *
+     * @param {ChatRoom} room the main room.
+     */
+    constructor(room: ChatRoom) {
+        this._mainRoom = room;
+        this._xmpp = room.xmpp;
+
+        this._handleMessages = this._handleMessages.bind(this);
+        this._mainRoom.xmpp.addListener(XMPPEvents.POLLS_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Whether polls is supported on backend.
+     *
+     * @returns {boolean} whether polls is supported on backend.
+     */
+    isSupported() {
+        return Boolean(this._xmpp.pollsComponentAddress);
+    }
+
+    /**
+     * Stops listening for events.
+     */
+    dispose() {
+        this._mainRoom.xmpp.removeListener(XMPPEvents.POLLS_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Creates and sends a new poll.
+     *
+     * @param pollId
+     * @param question
+     * @param answers
+     */
+    createPoll(pollId: string, question: string, answers: Array<{ name: string; }>) {
```

Comment:
Why not use an array of stringas at this level, rather than have to flatten it here?